code/modules/atmospherics/ZAS/ConnectionGroup.dm

```diff
@@ -0,0 +1,307 @@
+		A check against zas_settings.lightest_airflow_pressure should generally be performed before calling this.
+
+	get_connected_zone(zone/from)
+		Helper proc that allows getting the other zone of an edge given one of them.
+		Only on /connection_edge/zone, otherwise use A.
+
+*/
+
+
+/connection_edge
+	var/zone/A
+
+	var/list/connecting_turfs = list()
+	var/direct = 0
+	var/sleeping = 1
+	var/coefficient = 0
+
+	#ifdef ZASDBG
+	///Set this to TRUE during testing to get verbose debug information.
+	var/verbose = FALSE
+	#endif
+
+/connection_edge/New()
+	CRASH("Cannot make connection edge without specifications.")
+
+///Adds a connection to this edge. Usually increments the coefficient and adds a turf to connecting_turfs.
+/connection_edge/proc/add_connection(connection/c)
+	coefficient++
+	if(c.direct())
+		direct++
+
+	#ifdef ZASDBG
+	if(verbose)
+		zas_log("Connection added: [type] Coefficient: [coefficient]")
+	#endif
+
+///Removes a connection from this edge. This works even if c is not in the edge, so be careful.
+/connection_edge/proc/remove_connection(connection/c)
+	coefficient--
+	if(coefficient <= 0)
+		erase()
+	if(c.direct())
+		direct--
+
+	#ifdef ZASDBG
+	if(verbose)
+		zas_log("Connection removed: [type] Coefficient: [coefficient-1]")
+	#endif
+
+///Returns true if either A or B is equal to Z. Unsimulated connections return true only on A.
+/connection_edge/proc/contains_zone(zone/Z)
+	return
+
+///Removes this connection from processing and zone edge lists.
+/connection_edge/proc/erase()
+	SSzas.remove_edge(src)
+	#ifdef ZASDBG
+	if(verbose)
+		zas_log("[type] Erased.")
+	#endif
+
+///Called every air tick on edges in the processing list. Equalizes gas.
+/connection_edge/proc/tick()
+	return
+
+///Checks both connection ends to see if they need to process.
+/connection_edge/proc/recheck()
+	return
+
+///Airflow proc causing all objects in movable to be checked against a pressure differential. See file header for more info.
+/connection_edge/proc/flow(list/movable, differential, repelled)
+	for(var/i in 1 to length(movable))
+		var/atom/movable/M = movable[i]
+
+		//If they're already being tossed, don't do it again.
+		if(M.last_airflow > world.time - zas_settings.airflow_delay) continue
+		if(M.airflow_speed) continue
+
+		//Check for knocking people over
+		if(ismob(M) && differential > zas_settings.airflow_stun_pressure)
+			if(M:status_flags & GODMODE) continue
+			M:airflow_stun()
+
+		if(M.check_airflow_movable(differential))
+			//Check for things that are in range of the midpoint turfs.
+			var/list/close_turfs = list()
+			for(var/turf/U in connecting_turfs)
+				if(get_dist(M,U) < world.view) close_turfs += U
+			if(!close_turfs.len) continue
+
+			M.airflow_dest = pick(close_turfs) //Pick a random midpoint to fly towards.
+
+			//Soul code im too scared to touch
+			if(repelled) spawn if(M) M.RepelAirflowDest(differential/5)
+			else spawn if(M) M.GotoAirflowDest(differential/10)
```

Comment:
I am willing to bet these spawns existed to allow for sleep looping, but we have subsystems now

Reply:
I am very scared to change this, unironically.

Reply:
you know what spawn does, right?